openbb_energy/eia/crude_oil/crude_oil.py

```diff
@@ -0,0 +1,152 @@
+"""EIA Crude Oil Import Models."""
+
+import warnings
+from datetime import date as dateType
+from typing import Any, Dict, List, Literal, Optional
+
+from openbb_core.provider.abstract.data import Data
+from openbb_core.provider.abstract.fetcher import Fetcher
+from openbb_core.provider.abstract.query_params import QueryParams
+from pydantic import Field, field_validator
+
+_warn = warnings.warn
+
+CRUDE_OIL_FACET_LIST = ["originId", "originType", "destinationId", "destinationType", "gradeId"]
+
+
+class CrudeOilImportsQueryParams(QueryParams):
+    """Crude Oil Imports query parameters."""
+
+    frequency: Literal["monthly", "annual"] = Field(
+        description="Frequency of the data to be returned.",
+        default="monthly",
+    )
+
+    start_date: Optional[str] = Field(
+        description="Start date of the data to be returned."
+        + " Format: YYYY for annual data, YYYY-MM for monthly data.",
+        default=None,
+        alias="start",
+    )
+
+    end_date: Optional[str] = Field(
+        description="End date of the data to be returned."
+        + " Format: YYYY for annual data, YYYY-MM for monthly data.",
+        default=None,
+        alias="end",
+    )
+
+    filter_by_origin_id: Optional[str] = Field(
+        description="Filter by origin ID. You can provide a comma separated list of origin IDs."
+        + " Examples: CTY_CA (Canada), CTY_SA (Saudi Arabia), REG_ME (Middle East).",
+        default=None,
+        alias="originId",
+    )
+
+    filter_by_origin_type: Optional[str] = Field(
+        description="Filter by origin type. You can provide a comma separated list of origin types."
+        + " Choose from: REG (Region), CTY (Country), OPN (OPEC/non-OPEC), WORLD (World).",
+        default=None,
+        alias="originType",
+    )
+
+    filter_by_destination_id: Optional[str] = Field(
+        description="Filter by destination ID. You can provide a comma separated list of destination IDs."
+        + " Examples: RF_394 (refinery), PT_1003 (port), PS_LA (state).",
+        default=None,
+        alias="destinationId",
+    )
+
+    filter_by_destination_type: Optional[str] = Field(
```

Comment:
Replace Optional[str] with Optional[Literal[...]] that specifies the available options.